azext_iot/_factory.py

```diff
@@ -75,9 +76,13 @@ def __init__(self, target, device_id=None, auth_override=None):
         self.auth_override = auth_override
 
         # This initialization will likely need to change to support more variation of SDK
-        self.sas_uri = self.target["entity"]
+        self.sas_uri = self.target.get("entity", IOTDPS_PROVISIONING_HOST)
```

Comment:
It feels weird to have IOTDPS_PROVISIONING_HOST used for the target endpoint if no entity value exists when there are no additional conditions.

Reply:
These changes may be leftover changes from the generated sdk. Will check and remove if needed. 

Otherwise will add an optional parameter to make this more customizable (if there are multiple default sas_uri options, like maybe one for hub)